crates/hir-def/src/item_tree/lower.rs

```diff
@@ -411,6 +410,7 @@ impl UseTreeLowering<'_> {
         tree: ast::UseTree,
         span_for_range: &mut dyn FnMut(::tt::TextRange) -> SyntaxContext,
     ) -> Option<UseTree> {
+        self.mapping.alloc(tree.clone());
```

Comment:
That early allocation could cause problems because we return `None` in some cases, so the mapping will be corrupted.